client/electron/scripts/sign-latest-release.py

```diff
@@ -0,0 +1,137 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+"""
+Automate the signing of the latest release artifact.
+
+This script downloads the latest pre-built artifact from the GitHub repository,
+unzips it, checks the version, and create a signed windows installer using the
+embedded signing script.
+
+
+Requirements:
+- Python 3.9+
+- gh (GitHub CLI) installed and authenticated
+- tqdm installed (pip install tqdm)
+- urllib3>=2 installed (pip install "urllib3>=2")
+"""
+
+from __future__ import annotations
+
+
+import argparse
+import json
+import pathlib
+import shutil
+import subprocess
+from zipfile import ZipFile
+
+import tqdm
+import urllib3
+
+SIGN_SCRIPT = "sign-windows-package.cmd"
+ARTIFACT_NAME = "windows-exe-X64-electron-pre-built"
+BASE_URL = "https://api.github.com/repos/scille/parsec-cloud"
+
+
+def get_github_token() -> str:
+    process = subprocess.run(
+        "gh auth token", capture_output=True, shell=True, check=True, text=True
+    )
+    return process.stdout.strip()
+
+
+def get(url: str, token: str) -> urllib3.HTTPResponse:
+    headers = {
+        "Accept": "application/vnd.github+json",
+        "Authorization": f"Bearer {token}",
+        "X-GitHub-Api-Version": "2022-11-28",
+    }
+    return urllib3.request("GET", url, headers=headers, preload_content=False)  # type: ignore
+
+
+def get_artifact(
+    base_url: str, token: str, version: None | str = None
+) -> tuple[str, int, str, str, str]:
+    url = f"{base_url}/actions/artifacts"
+    artefacts = get(url, token).json()["artifacts"]  # type: ignore
```

Comment:
Pedantic mode: you are mixing "art**i**facts" (US spelling) and "art**e**facts" (UK spelling). I prefer artifact, but [both are accepted](https://english.stackexchange.com/a/37904). BTW GitHub calls them "assets" :sweat_smile:

Reply:
Github uses for its API btw: https://docs.github.com/en/rest/actions/artifacts

Reply:
b7a861c8fd56a84435a88315563c128a4797b220